### ISSUE TITLE
InitializerDeclaration cannot have annotation.

### ISSUE THREAD

**Issue by thanhminhmr** [javaparser/javaparser#2588]
Bear with me if I'm wrong, but a initializer block cannot have annotation, so `com.github.javaparser.ast.body.InitializerDeclaration.getAnnotations()` should not exist or should throw, isn't it?

**Comment by edefazio** [comment]
Yes, thats one of those outliers because `InitializerDeclaration` extends `BodyDeclaration` which implements `NodeWithAnnotations<T>`

```java
public abstract class BodyDeclaration<T extends BodyDeclaration<?>> extends Node implements NodeWithAnnotations<T> 
```

...and in all other situations `InitializerDeclaration` behaves like and is categorized as a `BodyDeclaration` so... yes it's probably not ideal.

(Maybe in the future we can make the individual `BodyDeclaration`s implement `NodeWithAnnotation` but for now... its just a "feature")

cheers

**Comment by edefazio** [comment]
After more thought, I don't think itll be too easy of a change (it could break alot of existing user-code)
so... Imma close this unless someone has an objection to it.

**Comment by thanhminhmr** [comment]
I do read the source code and notice that. Personally I think throw an exception here and at the set method would be okay, as it would minimize the damage to already existed code, and those fixed methods will behave like resolve FieldDeclaration that has multiple VariableDeclarator (throw on call). I don't like the idea of generating an uncompilable code from the AST tree.

> 
> 
> Yes, thats one of those outliers because `InitializerDeclaration` extends `BodyDeclaration` which implements `NodeWithAnnotations<T>`
> 
> ```java
> public abstract class BodyDeclaration<T extends BodyDeclaration<?>> extends Node implements NodeWithAnnotations<T> 
> ```
> 
> ...and in all other situations `InitializerDeclaration` behaves like and is categorized as a `BodyDeclaration` so... yes it's probably not ideal.
> 
> (Maybe in the future we can make the individual `BodyDeclaration`s implement `NodeWithAnnotation` but for now... its just a "feature")
> 
> cheers

**Comment by edefazio** [comment]
I'll reopen if you want to argue its necessity (here or on the gitter channel)
and will put in the work to update the JavaParser code & tests write tests for this code 
(or find someone/group willing to do so)

I closed it because (based on my experience) the juice isn't worth the squeeze...

Perhaps there are many users irritated at the API that find this important, so feel free to explain why...

In short, I'd hate for code people have written (perhaps a while ago) that uses the assumption that all `BodyDeclaration`s have annotations to start suddenly failing because of this change. 

(`InitializerDeclaration`s aren't used nearly as often as other `BodyDeclaration`s)

Thats my 2 cents,

cheers
